pkg/gofr/handler.go

```diff
@@ -149,8 +149,30 @@ func panicRecoveryHandler(re any, log logging.Logger, panicked chan struct{}) {
 func (h handler) logError(traceID string, err error) {
 	if err != nil {
 		errorLog := &ErrorLogEntry{TraceID: traceID, Error: err.Error()}
-		h.container.Logger.Error(errorLog)
+
+		if isHTTPError(err) {
+			h.container.Logger.Info(errorLog)
+		} else {
+			h.container.Logger.Error(errorLog)
+		}
+	}
+}
+
+//nolint:govet // We are using this function to check if the error is of type HTTP error.
+func isHTTPError(err error) bool {
+	httpErrors := []error{
+		&gofrHTTP.ErrorEntityAlreadyExist{},
+		&gofrHTTP.ErrorEntityNotFound{},
+		&gofrHTTP.ErrorInvalidParam{},
+		&gofrHTTP.ErrorMissingParam{},
+	}
+	for _, knownErr := range httpErrors {
+		if errors.As(err, &knownErr) {
+			return true
+		}
 	}
+
+	return false
 }
```

Comment:
If you have to do this, it means you could do better with the struct definition

```suggestion
func isHTTPError(err error) bool {
	return errors.Is(err, gofrHTTP.Error) {
}
```

Now let's move to gofr http package

Here is the current code for ErrorEntityNotFound

```go
// ErrorEntityNotFound represents an error for when an entity is not found in the system.
type ErrorEntityNotFound struct {
        Name  string
        Value string
}

func (e ErrorEntityNotFound) Error() string {
        // For ex: "No entity found with id: 2"
        return fmt.Sprintf("No entity found with %s: %s", e.Name, e.Value)
}

func (ErrorEntityNotFound) StatusCode() int {
        return http.StatusNotFound
}
```

Let's define a error, then add a method to ErrorEntityNotFound

```go
var ErrCommon = errors.New("HTTP error")

func (ErrorEntityNotFound) UnWrap() error {
        return ErrCommon
}
```

Reply:
You changed the implementation

Please consider what I suggested anyway

Reply:
You can even define errCommon as unexported

And add this to http package

```go
func IsHTTPError(err error) bool {
    return errors.Is(err, errCommon)
}
```

Reply:
@ccoVeille we can't use `errors.Is` as it will match the exact value (then the populated fields in the error struct will be treated as different error)

So based on your comment can we do something like this:

```go
import (
    "errors"
    gofrHTTP "path/to/gofrHTTP"
)

// Log the error (if any) with traceID and errorMessage.
func (h handler) logError(traceID string, err error) {
    if err != nil {
        errorLog := &ErrorLogEntry{TraceID: traceID, Error: err.Error()}

        if isKnownError(err) {
            h.container.Logger.Info(errorLog)
        } else {
            h.container.Logger.Error(errorLog)
        }
    }
}

func isKnownError(err error) bool {
    switch {
    case isEntityAlreadyExistError(err):
        return true
    case isEntityNotFoundError(err):
        return true
    case isInvalidParamError(err):
        return true
    case isMissingParamError(err):
        return true
    default:
        return false
    }
}

func isEntityAlreadyExistError(err error) bool {
    var target *gofrHTTP.ErrorEntityAlreadyExist
    return errors.As(err, &target)
}

func isEntityNotFoundError(err error) bool {
    var target *gofrHTTP.ErrorEntityNotFound
    return errors.As(err, &target)
}

func isInvalidParamError(err error) bool {
    var target *gofrHTTP.ErrorInvalidParam
    return errors.As(err, &target)
}

func isMissingParamError(err error) bool {
    var target *gofrHTTP.ErrorMissingParam
    return errors.As(err, &target)
}

```

Reply:
I disagree, mainly because that's the purpose of adding Unwrap

I could work on a better code, but I don't have time yet

Reply:
@ccoVeille will wait for your suggestion on a better code.

Reply:
I opened #1298